spec/factories/messages.rb

```diff
@@ -31,7 +31,11 @@
   end
 
   trait :with_attachment do
-    attachment { File.open(Rails.root + 'features/examples/shorter_lorem.docx') }
-    attachment_content_type { 'application/msword' }
+    attachment do
+      Rack::Test::UploadedFile.new(
+        File.expand_path('features/examples/shorter_lorem.docx', Rails.root),
+        'application/msword'
+      )
```

Comment:
Not sure what is preferable but we are using `fixture_file_upload` in other factories and specs. it is [included globally](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/blob/master/spec/rails_helper.rb#L71). However i have seen older comments suggesting it is bad practice to use this.

[example discussion - see comments](https://collectiveidea.com/blog/archives/2017/01/16/testing-an-uploaded-file-in-rspec)

Reply:
I have [PR'd a change to not use fixture_file_upload](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3835) to see how it goes

Reply:
I hadn't specifically avoided using `fixture_file_upload` and I don't know about the suggestion of it being bad practice. If we are moving away from `fixture_file_upload` are you happy for this to stay as it is?